buildcontext/gitlookup.go

```diff
@@ -615,26 +617,33 @@ func parseGitProtocol(remote string) (string, int) {
 //	"github.com/earthly/earthly/examples/go" ---> ("git@github.com/earthly/earthly.git", "examples/go")
 //
 // Additionally a ssh keyscan might be returned (or an empty string indicating none was configured)
-func (gl *GitLookup) GetCloneURL(path string) (string, string, []string, error) {
+// Also, a custom "git ssh command" may be returned. This is part of this function since the user may
+// specify a command necessary to clone their repository successfully.
+func (gl *GitLookup) GetCloneURL(path string) (gitURL string, subPath string, keyScans []string, sshCommand string, err error) {
 	gl.mu.Lock()
 	defer gl.mu.Unlock()
 	match, m, err := gl.getGitMatcherByPath(path)
 	if err != nil {
-		return "", "", nil, err
+		return "", "", nil, "", err
 	}
 
 	n := len(match)
-	subPath := ""
+	subPath = ""
 	if len(path) > n {
 		subPath = path[n+1:]
 		path = path[:n]
 	}
 	host := path[:strings.IndexByte(path, '/')]
 	gitPath := m.prefix + match[(strings.IndexByte(match, '/')+1):] + m.suffix
 
+	sshCommand = ""
+	if m.sshCommand != "" {
+		sshCommand = m.sshCommand
```

Comment:
very cool to see this was done in a way that allows for custom commands for particular hosts.